### PR TITLE
OpenAPI: Add endpoint field to CatalogConfig

### PR DESCRIPTION
**Pull request by nastra**

### CODE REVIEW

File: open-api/rest-catalog-open-api.yaml [hunk 1]
```diff
@@ -114,7 +117,13 @@ paths:
                 },
                 "defaults": {
                   "clients": "4"
-                }
+                },
+                "endpoints": [ "GET /v1/{prefix}/namespaces/{namespace}",
+                               "GET /v1/{prefix}/namespaces",
```
Comment (danielcweeks) [inline open-api/rest-catalog-open-api.yaml:122]: Formatting is a little awkward here.  Maybe start the list entries on a new line?

**Review by danielcweeks — APPROVED** [review]

**Review by amogh-jahagirdar — COMMENTED** [review]
Had the same comment as @rdblue here https://github.com/apache/iceberg/pull/10928/files#r1715618702

I think that spec needs to cover what endpoints can be assumed to be supported if nothing is returned. That ensures a compatibility in the protocol that clients can make use of.

Other than that, the changes LGTM.

File: open-api/rest-catalog-open-api.yaml (hunk 1 above)
Comment (nastra) [reply open-api/rest-catalog-open-api.yaml:122]: done

**Review by flyrain — APPROVED** [review]
+1. Thanks @nastra for driving this!

**Pull request merged**